editor/editor_node.cpp

```diff
@@ -4575,15 +4575,19 @@ bool EditorNode::is_object_of_custom_type(const Object *p_object, const StringNa
 }
 
 void EditorNode::progress_add_task(const String &p_task, const String &p_label, int p_steps, bool p_can_cancel) {
-	if (singleton->cmdline_export_mode) {
+	if (!singleton) {
+		return;
+	} else if (singleton->cmdline_export_mode) {
```

Comment:
I don't understand what bug this is fixing: it should be in a separate PR fixing a documented issue.

I don't think I can review this editor node code properly when combined with this change.

Reply:
This is for the unit test to work - the import process is trying to post progress messages to editor, but there is no editor singleton in a test.

I've talked a bit about it [here](https://github.com/godotengine/godot/pull/86183#issuecomment-1859448355), I didn't found any import/export tests to take inspiration from, so hacked it together and this seemed like the cleanest solution.

Reply:
I looked into instantiating the editor in the unit test, but it's not easy. There's no other test that does that, so lots of the assumptions the editor code makes do not hold true (similar to the one we're discussing here).

I could remove the unit test to avoid making changes to the editor_node, if that's preferable.

Reply:
yeah I appreciate adding the test and understand that mocking singletons is difficult: I don't know a better way to do this. Thanks for looking into it!